lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -1616,6 +1618,631 @@ class ConvertAtenAdaptivePoolOp : public OpConversionPattern<OpTy> {
 };
 } // namespace
 
+namespace {
+template <typename OpTy, typename PoolingOpTy, int Dim>
+class ConvertRoiAlignOp : public OpConversionPattern<OpTy> {
+public:
+  using OpConversionPattern<OpTy>::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(OpTy op, typename OpTy::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    Location loc = op->getLoc();
+    const TypeConverter *typeConverter = this->getTypeConverter();
+    Value result = op.getResult();
+
+    uint64_t pooledHeight =
+        cast<ConstantIntOp>(op.getPooledHeight().getDefiningOp()).getValue();
+    uint64_t pooledWidth =
+        cast<ConstantIntOp>(op.getPooledWidth().getDefiningOp()).getValue();
+    uint64_t samplingRatio =
+        cast<ConstantIntOp>(op.getSamplingRatio().getDefiningOp()).getValue();
+    Value pooledH = op.getPooledHeight();
+    Value pooledW = op.getPooledWidth();
+    Value spatialScaleVal = op.getSpatialScale();
+    llvm::APFloat spatialScale =
+        cast<ConstantFloatOp>(op.getSpatialScale().getDefiningOp()).getValue();
+    Value rois = op.getRois();
+    Value input = op.getInput();
+    // RankedTensorType inputType = input.getType();
+    Value offset =
+        rewriter.create<arith::ConstantOp>(loc, b.getF32FloatAttr(0.0));
+    Type resultType = cast<RankedTensorType>(result.getType());
+    Type resultElementType = resultType.getElementType();
+    if (!op.getAligned()) {
+      offset = rewriter.create<arith::ConstantOp>(loc, b.getF32FloatAttr(0.5));
+    }
+
+    Value lb = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    Value ub0 = rewriter.create<tensor::DimOp>(loc, rois, 0);
+    Value ub1 = rewriter.create<tensor::DimOp>(loc, input, 1);
+    Value step = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+    SmallVector<Value> finalOutputShape = {ub0, ub1, pooledH, pooledW};
+    Value finalOutputTensor = rewriter.create<tensor::EmptyOp>(
+        loc, getAsOpFoldResult(finalOutputShape), resultElementType);
+    auto forLoop = rewriter.create<scf::ForOp>(
+        loc, lb, ub0, step, ValueRange{},
+        [&](OpBuilder &b1, Location loc, Value iv0, ValueRange args) {
+          auto forLoop = b1.create<scf::ForOp>(
+              loc, lb, ub1, step, ValueRange{},
+              [&](OpBuilder &b, Location loc, Value iv1, ValueRange args) {
+                // Step 1: Extract bounds for region of interest (roi)
```

Comment:
nit: Use proper punctuation in comments
```suggestion
                // Step 1: Extract bounds for region of interest (roi).
```
See https://llvm.org/docs/CodingStandards.html#commenting

Reply:
Also in other comments in this file